sys/log/full/src/log.c

```diff
@@ -955,18 +955,17 @@ log_read_hdr(struct log *log, const void *dptr, struct log_entry_hdr *hdr)
 {
     int bytes_read;
 
+#if MYNEWT_VAL(LOG_FLAGS_IMAGE_HASH)
+    bytes_read = log_read(log, dptr, hdr, 0, (LOG_BASE_ENTRY_HDR_SIZE + LOG_IMG_HASHLEN));
+    if (bytes_read != (LOG_BASE_ENTRY_HDR_SIZE + LOG_IMG_HASHLEN)) {
+        return SYS_EIO;
+    }
+#else
     bytes_read = log_read(log, dptr, hdr, 0, LOG_BASE_ENTRY_HDR_SIZE);
     if (bytes_read != LOG_BASE_ENTRY_HDR_SIZE) {
         return SYS_EIO;
     }
-
-    if (hdr->ue_flags & LOG_FLAGS_IMG_HASH) {
```

Comment:
I think this is going to cause issues, the image hash was part of the header for backwards compatibility. Removing it might cause issues for other people.

Reply:
Thanks Vipul. The gains from this is less compared to the other change, so updated the PR and dropped this change. The unit tests should be probably expanded to ensure the backward compatibility use-cases and this change could be revisited.